### PR TITLE
[Implements #43] Add Tool Dependencies

### PR DESCRIPTION
**Pull request by cascadianblue**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ services:
       - 9001:8080
 
   physical-address-annotator:
-    image: nlpsandbox/physical-address-annotator-example:0.3.1
+    image: nlpsandbox/physical-address-annotator-example:edge
```
Comment (cascadianblue) [inline docker-compose.yml:19]: The `0.3.1` version tag of this annotator doesn't seem to implement the `tool` endpoint

Comment (tschaffter) [reply docker-compose.yml:19]: Strange. But yes, the endpoint `/tool` is in edge. I will release shortly a version 0.3.2 of the annotators shortly and then let you know when you can rely on the new versions.

**Comment by cascadianblue** [comment]
Hold on, I need to fix the unit test for this

File: server/openapi_server/controllers/tool_controller.py [hunk 2]
```diff
@@ -12,7 +14,7 @@ def get_tool():  # noqa: E501
     tool = Tool(
         name="phi-deidentifier",
         version="0.3.1",
-        license="apache-2.0",
+        license=License.APACHE_2_0,
```
Comment (tschaffter) [inline server/openapi_server/controllers/tool_controller.py:17]: Nice!

File: server/openapi_server/controllers/tool_controller.py [hunk 3]
```diff
@@ -31,4 +33,4 @@ def get_tool_dependencies():  # noqa: E501
 
     :rtype: ToolDependencies
     """
-    return [], 200
+    return annotator_client.get_annotators_info(), 200
```
Comment (tschaffter) [inline server/openapi_server/controllers/tool_controller.py:36]: The variable `annotator_client` is confusing. Would the name `deidentifier.get_dependencies_info()` be more suitable?

File: server/openapi_server/utils/annotator_client.py [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
-    'text_person_name': 'http://person-name-annotator:8080/api/v1/textPersonNameAnnotations',
-    'text_physical_address': 'http://physical-address-annotator:8080/api/v1/textPhysicalAddressAnnotations'
+BASE_URLS = {
+    'text_date': Config().date_annotator_api_url,
+    'text_person_name': Config().person_name_annotator_api_url,
+    'text_physical_address': Config().physical_address_annotator_api_url
 }
+
+
 CAMEL_CASE = {
```
Comment (tschaffter) [inline server/openapi_server/utils/annotator_client.py:13]: Replace by a descriptive variable

File: server/openapi_server/utils/annotator_client.py [hunk 5]
```diff
@@ -1,26 +1,43 @@
...
     'text_date': 'textDateAnnotations',
     'text_person_name': 'textPersonNameAnnotations',
     'text_physical_address': 'textPhysicalAddressAnnotations'
 }
 
 
+TOOL_URL = 'tool'
```
Comment (tschaffter) [inline server/openapi_server/utils/annotator_client.py:20]: Rename variable (the value does not look like a URL)

**Review by tschaffter — CHANGES_REQUESTED** [review]

File: server/openapi_server/utils/annotators.py [hunk 6]
```diff
@@ -10,30 +10,30 @@
...
-CAMEL_CASE = {
+ANNOTATION_TYPES_CAMEL_CASE = {
     'text_date': 'textDateAnnotations',
     'text_person_name': 'textPersonNameAnnotations',
     'text_physical_address': 'textPhysicalAddressAnnotations'
 }
 
 
-TOOL_URL = 'tool'
+TOOL_URL_SUFFIX = 'tool'
```
Comment (tschaffter) [inline server/openapi_server/utils/annotators.py:20]: Better: `TOOL_URL_PATH = '/tool'`

**Review by tschaffter — APPROVED** [review]
Looks great!

**Pull request merged**